woocommerce/woocommerce
# Fix various wc download handler issues

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:





Closes #51561 
Closes #51562  
Closes #51563

### How to test the changes in this Pull Request:

See the closed issues. You can use the existing tests, since the changes mostly affect the interface between PHP and the server.
There is no change to the end-user.

### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
WC_Download_Handler buffer handling
WC_Download_Handler error default status set to 403 to prevent files from not being served if 404 are cached by a page cache and to ensure the PHP status header matches the downstream reverse-proxy status

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
